cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -143,23 +159,27 @@ spec:
             - "c7in"
             - "m7in"
             - "r7in"
-        - key: "karpenter.k8s.aws/instance-size"
-          operator: "NotIn"
-          values:
-            - "metal"
-        # exclude instance-types with slow SSD
-        - key: "node.kubernetes.io/instance-type"
-          operator: "NotIn"
-          values:
-            - "c5d.large"
-#{{ else }}
+#{{ end }}
+#{{ if eq .NodePool.KarpenterInstanceTypeStrategy "custom" }}
         - key: "node.kubernetes.io/instance-type"
           operator: In
           values:
 #      {{ range $type := .NodePool.InstanceTypes }}
           - "{{ $type }}"
 #      {{ end }}
 #{{ end }}
+        # user's custom configurations
+#{{ if (index .NodePool.ConfigItems "requirements") }}
+#    {{ range $requirement := .NodePool.KarpenterRequirements }}
+        - key: "{{ $requirement.Key }}"
+          operator: "{{ $requirement.Operator }}"
+          values:
+#         {{ range $value := $requirement.Values }}
+            - "{{ $value}}"
+#         {{ end }}
+#    {{ end }}
+#{{ end }}
+        # other configuration
```

Comment:
Are these commented lines here for documentation purpose? It looks a little confusing, maybe we can just put an example configuration in a large comment block above this? just to seperate out the documenting comments and the actual configuration making it easier to read?

Reply:
things like
        # other configuration
        # user's custom configurations
are definately comments for cotext

however, `#    {{ range $requirement := .NodePool.KarpenterRequirements }}` and so on, are no comments. they are functional go templates.

they are comments in the yaml, but go templates can render them without a problem. this way I can get the IDE to parse the yaml file correctly

Reply:
Just looks a bit confusing, maybe we can remove them later when we merge, so it doesn't look confusing like this?